src/control/lib/control/response.go

```diff
@@ -262,7 +262,7 @@ func convertMSResponse(ur *UnaryResponse, out interface{}) error {
 	msResp, err := ur.getMSResponse()
 	if err != nil {
 		if IsConnErr(err) {
-			return errMSConnectionFailure
+			return errors.Wrap(err, errMSConnectionFailure.Error())
```

Comment:
Hmm. I think this is backward. The message argument annotates the error, to provide context. So we want to return `errMSConnectionFailure` to let the caller know that the call failed due to a MS connection failure, and we annotate that with more detail about why the connection failed. Aside from that, we also need to be careful when changing what gets returned from these functions, because often the named/sentinel errors are used to drive behavior (e.g. retry logic).

Reply:
Good point. I adjusted the agent to look for the MS connection failure. I do think it's helpful to log what exactly happened there as an annotation.